Implement basic functions

somedd opened the pull request:
step1의 기능을 구현하였습니다.

somedd committed:
Implement basic functions

godrm commented on LadderGame/LadderGame/main.swift:
```diff
@@ -2,9 +2,87 @@
+    var completeLadder = ladderWithVerticalLine
+    for indexOfHeight in 0...(completeLadder.count - 1) {
+        for indexOfBlock in 1...(completeLadder[indexOfHeight].count - 1) {
+            if indexOfBlock % 2 == 1 {
```
`|` 표시하는 VerticalLine 은 항상 출력하는 값이니까 2차원 배열에 넣지 않는게 좋습니다.
항상 출력되는 값을 배열에 넣다보니 % 2 == 1 비교하는 블록이 생기고 복잡해졌네요. 

범위를 나타낼때도 `1...(completeLadder[indexOfHeight].count - 1)` 보다는 `1..<completeLadder[indexOfHeight].count` 형태로 -1을 사용하지 않으면서 명확하게 쓰는게 좋습니다.

godrm commented on LadderGame/LadderGame/main.swift:
```diff
@@ -2,9 +2,87 @@
+    print("최대 사다리 높이는 몇 개인가요?")
+    let heightOfLadder = readLine()
+    let ladderWithVerticalLine = drawVerticalLine(entryVal: Int(entry!)!, heightVal: Int(heightOfLadder!)!)
+    printShapeOfLadder(ladderSet: drawHorizontalLine(ladderWithVerticalLine: ladderWithVerticalLine))
```
(데이터/값을 채우는 함수) -> [데이터] -> (데이터/값을 출력하는 함수) 형태로 작성한 것은 잘 하셨습니다.
간단한 함수는 이렇게 겹쳐서(nested) 호출해도 되지만, 코드를 읽을 때 뒷부분 부터 읽어야 하기 때문에
바로 윗줄처럼 분리해서 명시적으로 어떤 변수를 만들어서 넘겨주는 것도 좋습니다. 참고하세요.

godrm reviewed (commented):
코멘트 남긴 사항을 보고 수정해보세요.

somedd committed:
Edit some functions

1. 세로줄을 넣는 함수를 삭제하였습니다.
2. for문 범위에 대한 표현을 보다 명확히 하였습니다.
3. 함수가 겹치는 부분을 변수로 넘겨주도록 수정하였습니다.

godrm commented on LadderGame/LadderGame/main.swift:
```diff
@@ -31,49 +11,61 @@ func makeRandomHorizontalLine () -> Bool {
+    guard randomVal == true else {
+         temp[ladder.indexOfHeight][ladder.indexOfHorizontalLine] = " "
+         return temp
+    }
```
guard는 이렇게 로직이 들어가서 위아래 모두 처리가 있는 경우는 적합하지 않습니다.
이런 경우는 if - else 가 더 읽기 좋을 것 같아요.

somedd replied:
넵! 감사합니다!

somedd committed:
Edit sources  of some functions

1. drawHorizontalLines와 drawOneHorizontalline함수 내 guard문을
개선하였습니다.

godrm commented on LadderGame/LadderGame/main.swift:
```diff
@@ -83,6 +75,8 @@ while runLadderGame == true {
     let heightOfLadder = readLine()
-    let ladderWithVerticalLine = drawVerticalLine(entryVal: Int(entry!)!, heightVal: Int(heightOfLadder!)!)
-    printShapeOfLadder(ladderSet: drawHorizontalLine(ladderWithVerticalLine: ladderWithVerticalLine))
+    let Ladder = GenerateSetOfLadder(entryVal: Int(entry!)!, heightVal: Int(heightOfLadder!)!)
```
Int(entry!)! 보다는 앞쪽에서 if let 혹은 guard let 처럼 옵셔널을 unwrapping 해서 처리하는 게 좋습니다.
Int(entry) 가 옵셔널을 리턴하는 경우에는 `Int(entry) ?? 0`형태로 초기값을 지정해 주는게 깔끔합니다.
가능하면 `!` 연산자는 직접 붙여서 쓰지마세요

somedd replied:
넵!

godrm commented on LadderGame/LadderGame/main.swift:
```diff
@@ -31,49 +11,61 @@ func makeRandomHorizontalLine () -> Bool {
+func GenerateSetOfLadder (entryVal : Int, heightVal : Int) -> [[String]] {
+    let setOfLadder : [[String]] = Array(repeatElement(Array(repeatElement("", count: entryVal - 1)), count: heightVal))
+    return setOfLadder
+}
```
함수명 이름은 GenerateSetOfLadder() 보다는 generateSetOfLadder() 가 좋습니다. 꼭 소문자를 쓰세요.

somedd replied:
넵!

godrm commented on LadderGame/LadderGame/main.swift:
```diff
@@ -31,49 +11,61 @@ func makeRandomHorizontalLine () -> Bool {
+                ladderWithHorizontalLine = drawOneHorizontalLine(randomVal: false,(ladderWithHorizontalLine, indexOfHeight, indexOfHorizontalLine))
+                continue
+            }
+            ladderWithHorizontalLine = drawOneHorizontalLine(randomVal: drawLineVal,(ladderWithHorizontalLine, indexOfHeight, indexOfHorizontalLine))
```
drawOneHorizontalLine() 함수의 역할은 무엇인가요?
2차원 배열에 값을 하나 채우는 동작인데 전체 2차원 배열을 다 넘기고 받는 것 같아 보입니다. 
사다리 하나 값만 활용할 수는 없는걸까요?

somedd replied:
넵 수정완료하였습니다!

somedd committed:
Reduce sources of some functions

1. drawOneHorizontalLine 함수내부 매개변수 타입과 리턴값을 바꿨습니다.
2. 함수명의 대소문자 구분하였습니다.
3. 메인부분의 옵셔널 unwrapping에 대한 처리를 하였습니다.

godrm commented:
수고하셨습니다. 다음 단계에 도전하세요 :) 
꼭 본인 이름 브랜치에서 fetch-rebase (또는 pull) 하고 push 하고 새 작업 브랜치를 만들고 시작하세요.
